.github/workflows/main_template.yml

```diff
@@ -1,41 +1,41 @@
 on:
-    workflow_call:
-        inputs:
-            modulepath:
-              required: true
-              type: string
-            tests-extra-args:
-              required: false
-              type: string
-        secrets:
-          codecov-token:
-            required: true
-
-# TODO: environment variables cannot be sent to reusable workflows: https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations
-# env:
-#   GO_VERSION: "1.22.x"
+  workflow_call:
+    inputs:
+      modulepath:
+        required: true
+        type: string
+      tests-extra-args:
+        required: false
+        type: string
+      go-version:
+        description: "Go version to use"
+        required: true
+        type: string
```

Comment:
I prefer to keep all references to go version inside only one file: this one.

Reply:
https://github.com/gnolang/gno/pull/3316#discussion_r1890359284